### PR TITLE
Add optional log callback to init function

### PR DESCRIPTION
**Pull request by hikinggrass**
Here a allback function can be registered that receives a LogRecord, containing the message, severity, timestamp, process and function (if available respectively)

### CODE REVIEW

File: examples/main.cpp [hunk 1]
```diff
@@ -28,7 +29,27 @@ int main(int argc, char* argv[]) {
     if (vm.count("logconf") != 0) {
         logging_config = vm["logconf"].as<std::string>();
     }
-    Everest::Logging::init(logging_config, "hello there");
+
+    if (vm.count("callback") != 0) {
+        std::cout << "Using log callback" << std::endl;
+        Everest::Logging::init(logging_config, "hello there", [](const Everest::Logging::LogRecord& record) {
+            std::cout << "Message: " << record.message << std::endl;
```
[the remark this answers is not in the document]
Comment (hikinggrass) [reply examples/main.cpp:36]: That's just for the example output, something that doesn't use the liblog logging

File: lib/logging.cpp [hunk 2]
```diff
@@ -92,11 +93,62 @@ std::istream& operator>>(std::istream& strm, severity_level& level) {
...
+        }
+        if (logging::value_ref<boost::posix_time::ptime> timestamp =
+                rec["TimeStamp"].extract<boost::posix_time::ptime>()) {
+            log_record.timestamp = boost::posix_time::to_iso_extended_string(*timestamp);
+            // TODO: conversion to chrono type?
+        }
+        if (logging::value_ref<std::string> process = rec["Process"].extract<std::string>()) {
+            log_record.process = *process;
+        }
+        if (logging::value_ref<std::string> function = rec["function"].extract<std::string>()) {
```
[the remark this answers is not in the document]
Comment (hikinggrass) [reply lib/logging.cpp]: I've added the line number and originating file as well

File: examples/main.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (maaikez) [reply examples/main.cpp:36]: Yeah we do use the std::cout logging to forward it currently... (oops)

**Review by maaikez — APPROVED** [review]

**Comment by hikinggrass** [comment]
This has been solved in a different way in libocpp

**Pull request closed without merging**